Clean up dependencies for sequoia-openpgp:

lalrpop-util: upgrade to 0.19 goes through NEW b/c of new "lexer"
         feature, Neal Walfield said nothing after 0.17 was used (and
         the lalrpop dependency itself is >=0.17)

drop all crypto-crng and windows stuff

And we fake out the "vendored-nettle" thing, since on debian,
nettle is linked dynamically

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,7 +56,7 @@
 version = "0.2"
 
 [dependencies.lalrpop-util]
-version = "0.19"
+version = ">=0.17"
 
 [dependencies.lazy_static]
 version = "1.4.0"
@@ -102,39 +102,10 @@
 compression = ["compression-deflate", "compression-bzip2"]
 compression-bzip2 = ["bzip2", "buffered-reader/compression-bzip2"]
 compression-deflate = ["flate2", "buffered-reader/compression-deflate"]
-crypto-cng = ["winapi", "win-crypto-ng", "ed25519-dalek", "num-bigint-dig"]
 crypto-nettle = ["nettle"]
 default = ["compression", "crypto-nettle"]
 vendored = ["vendored-nettle"]
-vendored-nettle = ["nettle/vendored"]
-[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.chrono]
-version = "0.4.10"
-features = ["std"]
-default-features = false
-[target."cfg(windows)".dependencies.eax]
-version = "0.3"
-
-[target."cfg(windows)".dependencies.ed25519-dalek]
-version = "1"
-features = ["rand", "u64_backend"]
-optional = true
-default-features = false
-
-[target."cfg(windows)".dependencies.num-bigint-dig]
-version = "0.6"
-optional = true
-default-features = false
-
-[target."cfg(windows)".dependencies.win-crypto-ng]
-version = "0.4"
-features = ["rand", "block-cipher"]
-optional = true
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.8"
-features = ["bcrypt"]
-optional = true
-default-features = false
+vendored-nettle = []
 [badges.gitlab]
 repository = "sequoia-pgp/sequoia"
 
